### PR TITLE
Stomp WebSocket example (for validating, not to merge yet)

### PR DESCRIPTION
**Pull request by thinhle19**

### CODE REVIEW

File: core/core-api/src/main/java/com/hbc/pms/core/api/task/PlcPublisher.java [hunk 1]
```diff
@@ -10,32 +11,35 @@
 @Component
 public class PlcPublisher {
 
-  private final Logger log = LoggerFactory.getLogger(getClass());
-  private final StationService stationService;
+    private final Logger log = LoggerFactory.getLogger(getClass());
```
Comment (huybui38) [inline core/core-api/src/main/java/com/hbc/pms/core/api/task/PlcPublisher.java:14]: use lombok instead

File: core/core-api/src/main/java/com/hbc/pms/core/api/config/WebSocketConfig.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker;
+import org.springframework.web.socket.config.annotation.StompEndpointRegistry;
+import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer;
+
+@Configuration
+@EnableWebSocketMessageBroker
+public class WebSocketConfig implements WebSocketMessageBrokerConfigurer {
+    @Override
+    public void registerStompEndpoints(StompEndpointRegistry registry) {
+        registry.addEndpoint("/websocket").setAllowedOrigins("*");
```
Comment (huybui38) [inline core/core-api/src/main/java/com/hbc/pms/core/api/config/WebSocketConfig.java:14]: We shouldn't allow all origins like this

File: core/core-api/src/main/java/com/hbc/pms/core/api/config/WebSocketConfig.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.simp.config.MessageBrokerRegistry;
+import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker;
+import org.springframework.web.socket.config.annotation.StompEndpointRegistry;
+import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer;
+
+@Configuration
+@EnableWebSocketMessageBroker
+public class WebSocketConfig implements WebSocketMessageBrokerConfigurer {
```
Comment (huybui38) [inline core/core-api/src/main/java/com/hbc/pms/core/api/config/WebSocketConfig.java:11]: Please document your findings in confluence. Things such as data the data flow, diagram etc... should be added to the document

**Review by huybui38 — CHANGES_REQUESTED** [review]

**Review by huybui38 — DISMISSED** [review]

File: core/core-api/src/main/java/com/hbc/pms/core/api/config/WebSocketConfig.java (hunk 3 above)
Comment (thinhle19) [reply core/core-api/src/main/java/com/hbc/pms/core/api/config/WebSocketConfig.java:11]: 1. Origin: I've configured the WebSocketConfig to allow the localhost origin only. I think these allowed origins should be put in a constant file, but I still don't know the appropriate path. I will update as soon as I can when find out.
2. Lombok: I've replaced the logger with the @Slf4j Lombok's annotation. And I see some places in the sample code still not using Lombok yet (e.g., PlcService.java, ApiControllerAdvice.java, etc.). Would you like me to update this also? 
3. Document: I mostly follow the two tuts mentioned here (https://hbc-capstone.atlassian.net/l/cp/krMW2bHc). But I also think documenting and diagraming the flow is a good idea for me to understand better. Regarding this, I will try to complete it before **midnight**

**Comment by oHTGo** [comment]
@thinhle19 please change PR title like #5

File: core/core-api/src/main/java/com/hbc/pms/core/api/config/WebSocketConfig.java (hunk 3 above)
Comment (thinhle19) [reply core/core-api/src/main/java/com/hbc/pms/core/api/config/WebSocketConfig.java:11]: https://hbc-capstone.atlassian.net/l/cp/028A1sT5
Hi @huybui38, I've captured the flow investigated in the code runtime. I really appreciate any of your comments, no matter how harsh they are, so I can understand more and produce better code. 
If everything looks fine with the sample code, I will discuss further with you and Kien the approach of handling multiple stations' data. 
Good night!

**Comment by oHTGo** [comment]
If the PR is not ready to merge, you should mark this pull request as a draft

**Pull request closed without merging**